Test/PublishNodeModule.Tests.ps1

```diff
@@ -297,13 +303,41 @@ Describe 'PublishNodeModule' {
 "@
         WhenPublishingNodeModule -WithCredentialID 'NpmCred' `
                                  -WithEmailAddress 'somebody@example.com' `
-                                 -WithNpmRegistryUri 'http://registry@example.com'
+                                 -WithNpmRegistryUri 'http://registry@example.com' `
+                                 -WithVersion '1.2.1'
         ThenNpmrcCreated -WithEmail 'somebody@example.com' -WithRegistry 'http://registry@example.com'
         ThenNpmPackagesPruned
         ThenNodeModulePublished
         ThenPublishedWithTag -None
     }
 
+    It 'should fail when module version hasn''t been updated.' {
+        GivenPackageJson @"
+        {
+            "name": "publishnodemodule_test",
+            "version": "1.2.0"
+        }
+"@
+        WhenPublishingNodeModule -WithCredentialID 'NpmCred' `
+                                 -WithEmailAddress 'somebody@example.com' `
+                                 -WithNpmRegistryUri 'http://registry@example.com' `
+                                 -ErrorAction SilentlyContinue
+        ThenTaskFailed 'failed with exit code 1'

```

Comment:
I don't understand this test. What version is expected/where does it come from? Why does this scenario fail the task?

Reply:
Look at the update I made to the `WhenPublishingNodeModule` function for the tests.

Previously the way the tests were set up, the version is retrieved from the package.json that's created from the `GivenPackageJson` function. I've added a new parameter `WithVersion` so we can explicitly set the version for the Whiskey Task Context.

I'll update the tests so that they all specify a version explicitly on the `WithVersion` parameter so that it's easier to understand.

Reply:
The test seems flawed, it's not representative of how the version would be resolved during a real build.

A `Version` task would read the version from the `package.json` file, which would mean the version in the TaskContext object matches the package.json version. But in the test, you're manually changing the TaskContext version to something else.

That's exactly why the test behaves in this way:

> Previously the way the tests were set up, the version is retrieved from the package.json that's created from the GivenPackageJson function.

because that's how the build would get its version.